hrms/hr/doctype/expense_claim/bulk_download.py

```diff
@@ -0,0 +1,192 @@
+import frappe
+from frappe import _
+from frappe.utils.csvutils import build_csv_response
+import pandas as pd
+from suite42_erp_app.suite42_erp_app.common.decorators import (
+    handle_exceptions_with_readable_message,
+)
+from suite42_erp_app.suite42_erp_app.common.authentication.reimbursement_constants import (
+    RoleConstants,
+    ExpenseClaimConstants,
+    EmployeeConstant,
+    CompanyConstants,
+)
+from suite42_erp_app.suite42_erp_app.common.utils import user_has_role
+from suite42_erp_app.overrides.custom_company.constants import CompanyNames
+import json
+from datetime import datetime
+
+
+@frappe.whitelist()
+@handle_exceptions_with_readable_message
+def bulk_download(company, bank):
+    if company != CompanyNames.SUITE42:
+        frappe.throw(_(f"Currently Only for Company {CompanyNames.SUITE42} is supported"))
+    elif bank != CompanyConstants.KOTAK_BANK:
+        frappe.throw(_(f"Currently Only for Bank {CompanyConstants.KOTAK_BANK} is supported "))
+
+    expense_claim_list = []
+    if not user_has_role(frappe.session.user, RoleConstants.FINANCE_ROLE):
+        frappe.throw(
+            _(f"Invalid permission, only user with role {RoleConstants.FINANCE_ROLE} is allowed")
+        )
+
+    employee_list = frappe.db.get_list(
+        "Employee",
+        filters={
+            "status": "active",
+            "employment_type": ["in", EmployeeConstant.EMPLOYEMENT_TYPE],
+            "company": CompanyNames.SUITE42,
+        },
+        pluck="name",
+        ignore_permissions=True,
+    )
+
+    for employee_name in employee_list:
+        pending_expense_claims = frappe.db.get_list(
+            "Expense Claim",
+            filters={
+                "employee": employee_name,
+                "status": ExpenseClaimConstants.PENDING_PAYMENT,
+                "grand_total": (">", 0),
+            },
+            fields=["name", "grand_total"],
+        )
+        if not pending_expense_claims:
+            continue
+
+        total_amount = 0
+        expense_claim_names = []
+        for row in pending_expense_claims:
+            expense_claim_names.append(row.get("name"))
+            total_amount += row.get("grand_total")
+
+        employee_doc = frappe.get_doc("Employee", employee_name)
+
+        # compay bank Account
+        company_bank_account_exists = frappe.db.get_list(
```

Comment:
take bank account as fn argument and use frappe.get_doc directly

Reply:
Done